hedera-node/docs/design/services/token-service/airdrops/airdrop-transaction.md

```diff
@@ -0,0 +1,208 @@
+     * "Pending Airdrop" MAY be canceled via a `cancelAirdrop` transaction.<br/>
+     * All transfer fees (including custom fees and royalties), as well as the rent cost for the
+     * first auto-renewal period for any automatic-association slot occupied by the airdropped
+     * tokens, SHALL be charged to the account submitting this transaction.
+     */
+    rpc airdropTokens (Transaction) returns (TransactionResponse);
+```
+
+### Fee types
+
+The airdrop transaction would include several types of fees, a different set of them would be applicable, based on the conditions the airdrop is performed to. The list of fees is:
+
+- Airdrop fee covering the operation itself
+- Association fee covering a pre-paid fee for associating the token being airdropped to the receiver, in case there is a missing relationship between the two
+- Custom fees covering any token custom fees, which are required to be paid by the sender of the potential transfer in case of claim
+- Transfer fee covering a pre-paid fee for transferring the token being airdropped to the receiver, so that when they potentially claim the token to be free of charge
+- Auto-renewal fees covering a pre-paid period for a full auto-renewal cycle
+- Fee covering auto-creation/lazy-creation of accounts, if we airdrop to an alias address of non-existing account
+
+Note that for all these fees, a given sender, which is the actual token owner, could be covered by a different payer, which is willing to pay. However, in case the receiver has not claimed an airdrop for a more than a full auto-renew cycle, the sender (token owner) would be required to continue paying future rents and they cannot be covered by a separate payer.
+
+An update into the `feeSchedule` file would be needed to specify the addition of the new fee values for the airdrop transaction type itself.
+
+### Keys
+
+Since a given sender/token owner could be the payer of the airdrop transaction or they can rely on a separate payer account we have a couple of use cases related to the key verification logic.
+
+If the sender of the airdrop is also the payer of this transaction, we would need only their key to be present in the transaction context.
+
+If the airdrop transaction fees from the list above are covered by a separate payer, then both the keys of the airdrop sender and the payer should be present in the context. That is needed, so that the sender authorizes giving an airdrop from their own balance.
+
+### Service updates
+
+- Update `TokenServiceDefinition` class to include the new RPC method definition for sending airdrops
+- Implement new `TokenAirdropHandler` class which should be invoked when the gRPC server handles `TokenAirdropTransaction` transactions. The class should be responsible for:
+    - Verify that the pending airdrops list contains between 1 and 10 entries, inclusive
+    - Verify that the pending airdrops list does not have any duplicate entries
```

Comment:
Do we disbar duplicates?  What if an airdrop smart contract were to send 3 tokens from Bob to Alice for using a new dApp, and send a bonus 3 tokens from Bob to Alice for "liking" the dApp account on `X`?